airflow/api_connexion/endpoints/task_instance_endpoint.py

```diff
@@ -248,7 +248,7 @@ def post_clear_task_instances(dag_id: str, session=None):
         error_message = f"Dag id {dag_id} not found"
         raise NotFound(error_message)
     reset_dag_runs = data.pop('reset_dag_runs')
-    task_instances = dag.clear(get_tis=True, **data)
+    task_instances = dag.clear(get_ti_instances=True, **data)
```

Comment:
I believe `tis` stands for `task instances`, so maybe `get_task_instances` to make it more explicit? 

Another problem, `ti` is also a "public" variable as it's used in xcom, task context, macros etc. Though in general, I prefer explicit over abbreviation, I don't think it's good change to include within this PR.

The same applies to the rest of `ti_instances`

Reply:
Ah, but what I had to do here was separate the two concepts of:

- TaskInstance model instances
- TaskInstance PK tuples

Since I had to re-write `dag.clear()` to remove a UNION, and wanted to make it slightly more efficient in the process, the internals of that function now work on PK tuples to deduplicate (rather than de-duplicating on all fields as previously), so it needed a new pair of keyword arguments to say what output you wanted. I felt `ti_instances` was a reasonable solution since it's both different to the old name (so I could make sure I had caught all references), and because they are actually TaskInstance (model) instances.

Reply:
Ah I see. I think these are some good reasons to move it to `get_ti_instances`, but I would still prefer to keep `get_tis` and add the `get_ti_keys`. 

> because they are actually TaskInstance (model) instances.

- Though it makes sense, I think it's a bit unconventional as normally people just reference the instances as lowercase to the model class name (i.e. `class User(Base)` -> `user = User(...)` ) as similar to an instance(object) of a class. 
- Another concern from me is that people in the future may feel confused about the naming when they come across this.

>  different to the old name (so I could make sure I had caught all references)

Would this cause breaking changes as `dag.clear()` is a public API? Though I doubt people actually use `dag.clear()` in writing the DAG, the method may still be used in testing. In this case, we still need to keep the old keyword with deprecation warnings.

WDYT? I am also curious about other people's opinions.

Reply:
`get_ti_objects` perhaps?

Reply:
https://github.com/apache/airflow/pull/15389#discussion_r638671367

Reply:
I'm going to rename the `instances` one back to `get_tis` since the tests have already run with the rename and not found any dangling references, and it should behave the same in 99% of cases - we can keep backwards compat here without too much issue.